cargo-concordium/src/main.rs

```diff
@@ -58,15 +59,23 @@ enum Command {
         )]
         state_bin_path: PathBuf,
     },
-    #[structopt(name = "test", about = "Build and run tests using a Wasm interpreter.")]
+    #[structopt(
+        name = "test",
+        about = "Build and run tests using a Wasm interpreter.
+This command also builds a deployable Wasm module for integration testing, and it therefore also \
+                 has arguments for building."
+    )]
     Test {
         #[structopt(name = "seed", long = "seed", help = "Seed for randomized testing")]
-        seed: Option<u64>,
+        seed:            Option<u64>,
+        #[structopt(flatten)]
+        build_options:   BuildOptions,
         #[structopt(
-            raw = true,
-            help = "Extra arguments passed to `cargo build` when building the test Wasm module."
+            name = "only-unit-tests",
+            long = "only-unit-tests",
+            help = "Whether only the unit tests should be run."
         )]
-        args: Vec<String>,
```

Comment:
I think we should keep this argument unless it causes problems with the `raw` argument of `BuildOptions`.

The normal `cargo test` allows you to say `cargo test foo` and it'll only run the tests that matches `foo` (not sure what "matches" is exactly, but is is somethlng like substring match).

If we kept this we could also do this. instead of adding all `--test` flags we would only add it to the tests that match.

Reply:
If just positional arguments don't work maybe we can use named arguments instead. The API won't be the same as cargo test, but I think that's acceptable if necessary.

Reply:
It is included in the `BuildOptions`, so it is still there. However, we could consider adding one set of args for `cargo build` and one set for `cargo test`. Right now the extra args are used for both, which might pose a problem.

Reply:
As discussed on slack, it is not possible to have two `raw` args, i.e. ones that capture all arguments after `cargo concordium test -- `, so we went for an option where the user can limit which test targets to run with the `--test` flag.

Reply:
Note that it is only possible to specify test _targets_ with `--test`. It won't try to match on the individual test names. So you need to split your tests into multiple files if you want the flexibility to run a subset of them.

Reply:
That's reasonable.